GCTC-NTGC/gc-digital-talent
# Debt - 3490 - Remove secondment field

Resolves #3490 

## Summary

This deprecates the `interestedInLaterOrSecondment` in the api and removes it from the frontend (profile and gov info form).

## Screenshot

<img width="1195" alt="Screen Shot 2022-08-17 at 1 39 59 PM" src="https://user-images.githubusercontent.com/4127998/185206200-5809140d-1153-4db6-b41c-035b38dd8a12.png">

## Testing

1. Run codegen `npm run codegen --workspace=talentsearch`
2. Build talentsearch `npm run production --workspace=talentsearch`
3. Login and navigate to **My Profile**
4. Click on "**Edit Government Information**" link
5. Confirm there is no question about lateral deployment or secondment when "**Do you currently work for the government of Canada?**" is set to "**Yes**"
